### PR TITLE
Update action version to 4

### PR DESCRIPTION
**Pull request by b10n1k**

### CODE REVIEW

File: .github/workflows/isotovideo-action.yml [hunk 1]
```diff
@@ -9,11 +9,9 @@ jobs:
...
-      - name: install jq
-        run: zypper -n in jq
 
       - name: Run isotovideo against test code
-        run: isotovideo qemu_no_kvm=1 casedir=.
+        run: isotovideo qemu_no_kvm=1 casedir=$(pwd)
 
       - name: fail if any test module failed
-        run: jq .result testresults/result-*.json | grep -v ok && echo "Test modules failed" && exit 1
+        run: jq .result testresults/result-*.json | grep ok || (echo "Test modules failed" && exit 1)
```
Comment (baierjan) [inline .github/workflows/isotovideo-action.yml:17]: Consider https://github.com/os-autoinst/os-autoinst-distri-example/pull/31#discussion_r1483208120

```suggestion
        run: jq .result testresults/result-*.json | grep -v ok && { echo "Test modules failed" && exit 1; } || true
```

File: .github/workflows/isotovideo-action.yml [hunk 2]
```diff
@@ -9,11 +9,9 @@ jobs:
       image: "registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86-jq"
     steps:
       - uses: actions/checkout@v2
-      - name: install jq
-        run: zypper -n in jq
 
       - name: Run isotovideo against test code
-        run: isotovideo qemu_no_kvm=1 casedir=.
+        run: isotovideo qemu_no_kvm=1 casedir=$(pwd)
```
Comment (okurz) [inline .github/workflows/isotovideo-action.yml:14]: why not keep .?

File: .github/workflows/isotovideo-check-all-test-modules.yml [hunk 3]
```diff
@@ -13,4 +13,9 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Run isotovideo against test code, fail if any test module failed
-        run: podman run --rm -it -v .:/tests:Z --entrypoint '' registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86-jq /bin/sh -c 'isotovideo qemu_no_kvm=1 casedir=/tests && jq .result testresults/result-*.json | grep -v ok && echo "Test modules failed" && exit 1'
+        run: |
+          docker run --rm -v .:/tests:Z --entrypoint '' \
+          registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86-jq /bin/sh \
+          -c 'isotovideo qemu_no_kvm=1 casedir=/tests \
+          && jq .result testresults/result-*.json | grep ok \
+          || (echo "Test modules failed" && exit 1)'
```
Comment (baierjan) [inline .github/workflows/isotovideo-check-all-test-modules.yml:21]: Same grep and subshell problem as above,
```suggestion
          && jq .result testresults/result-*.json | grep -v ok \
          && { echo "Test modules failed" && exit 1; } || true'
```

File: .github/workflows/isotovideo-check-all-test-modules.yml [hunk 4]
```diff
@@ -13,4 +13,9 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Run isotovideo against test code, fail if any test module failed
-        run: podman run --rm -it -v .:/tests:Z --entrypoint '' registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86-jq /bin/sh -c 'isotovideo qemu_no_kvm=1 casedir=/tests && jq .result testresults/result-*.json | grep -v ok && echo "Test modules failed" && exit 1'
+        run: |
+          docker run --rm -v .:/tests:Z --entrypoint '' \
```
Comment (baierjan) [inline .github/workflows/isotovideo-check-all-test-modules.yml:17]: Consider https://github.com/os-autoinst/os-autoinst-distri-example/pull/31#discussion_r1483167930

> Is the switch from podman to docker intended? I thought podman is the preferred way.

**Review by baierjan — CHANGES_REQUESTED** [review]

File: .github/workflows/isotovideo-action.yml (hunk 1 above)
Comment (okurz) [inline .github/workflows/isotovideo-action.yml:17]: that `grep ok` would just mean that there is at least one test module that failed. But the purpose was to fail if any module fails. As we expect the execution in the example to fail what we can do is

```suggestion
        run: jq .result testresults/result-*.json | grep -v ok && (echo "At least one test module failed as expected" && exit 0)
```

File: .github/workflows/isotovideo-check-all-test-modules.yml (hunk 3 above)
Comment (okurz) [inline .github/workflows/isotovideo-check-all-test-modules.yml:21]: same as above

File: .github/workflows/isotovideo.yml [hunk 5]
```diff
@@ -8,4 +8,4 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Run isotovideo against test code in happy-path scenario
-        run: podman run --rm -it -v .:/tests:Z registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86 qemu_no_kvm=1 casedir=/tests
+        run: docker run --rm -v .:/tests:Z registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86 qemu_no_kvm=1 casedir=/tests
```
Comment (okurz) [inline .github/workflows/isotovideo.yml:11]: Why would you change that?

**Review by okurz — CHANGES_REQUESTED** [review]

File: .github/workflows/isotovideo.yml (hunk 5 above)
Comment (b10n1k) [reply .github/workflows/isotovideo.yml:11]: podman is not provided in the container. I found it easier to just switch to docker.

File: .github/workflows/isotovideo-check-all-test-modules.yml (hunk 4 above)
Comment (b10n1k) [reply .github/workflows/isotovideo-check-all-test-modules.yml:17]: podman is not provided in the container. I found it easier to just switch to docker. i am not sure if that matters here

File: .github/workflows/isotovideo-check-all-test-modules.yml [hunk 6]
```diff
@@ -13,4 +13,9 @@ jobs:
     steps:
       - uses: actions/checkout@v2
```
Comment (baierjan) [inline .github/workflows/isotovideo-check-all-test-modules.yml:14]: https://github.com/os-autoinst/os-autoinst-distri-example/pull/31#discussion_r1483187814

> Can you also change the action to actions/checkout@v4?

File: .github/workflows/isotovideo-check-all-test-modules.yml (hunk 4 above)
Comment (okurz) [reply .github/workflows/isotovideo-check-all-test-modules.yml:17]: I did not observe that. See https://github.com/os-autoinst/os-autoinst-distri-example/actions/runs/7275772852/job/19825501448#step:3:1

File: .github/workflows/isotovideo-action.yml (hunk 2 above)
Comment (b10n1k) [reply .github/workflows/isotovideo-action.yml:14]: not a significant reason. personal preference. it s quite similar

Comment (okurz) [reply .github/workflows/isotovideo-action.yml]: then please revert

File: .github/workflows/isotovideo-action.yml (hunk 1 above)
Comment (okurz) [reply .github/workflows/isotovideo-action.yml]: Please see https://github.com/os-autoinst/os-autoinst-distri-example/pull/33#discussion_r1484061727

Comment (b10n1k) [reply .github/workflows/isotovideo-action.yml]: i applied suggestion from @baierjan. review and resolve please

File: .github/workflows/isotovideo-check-all-test-modules.yml (hunk 3 above)
Comment (b10n1k) [reply .github/workflows/isotovideo-check-all-test-modules.yml]: i applied suggestion from @baierjan. review and resolve please

File: .github/workflows/isotovideo-check-all-test-modules.yml (hunk 4 above)
Comment (baierjan) [reply .github/workflows/isotovideo-check-all-test-modules.yml:17]: Whops, I will close #35 in favor of yours, it seems we observed the same. 😄

File: .github/workflows/isotovideo-action.yml (hunk 1 above)
Comment (okurz) [reply .github/workflows/isotovideo-action.yml]: Yes, I saw that you applied the suggestion from @baierjan. The problem is that this way we are ignoring results either way. But we should check for the expected result which is actually that test modules *should* fail which serves as the example

File: .github/workflows/isotovideo-check-all-test-modules.yml (hunk 6 above)
Comment (b10n1k) [reply .github/workflows/isotovideo-check-all-test-modules.yml:14]: Sure. i was thinking about that. I just do not know where to find the latest version

Comment (baierjan) [reply .github/workflows/isotovideo-check-all-test-modules.yml:14]: https://github.com/actions/checkout

Comment (b10n1k) [reply .github/workflows/isotovideo-check-all-test-modules.yml:14]: Done

File: .github/workflows/isotovideo-action.yml [hunk 7]
```diff
@@ -9,11 +9,9 @@ jobs:
       image: "registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86-jq"
     steps:
       - uses: actions/checkout@v2
-      - name: install jq
-        run: zypper -n in jq
```
Comment (okurz) [inline .github/workflows/isotovideo-action.yml:13]: That's a good change but should be in a separate commit (does not have to be separate PR, just separate commit)

File: .github/workflows/isotovideo-action.yml (hunk 1 above)
Comment (baierjan) [reply .github/workflows/isotovideo-action.yml]: Do you suggest that it is expected that the test will fail (i.e. the result of the isotovideo is no ok)? Running the pipeline locally yields `testresults/result-boot.json` with the `"result" : "ok"`, so the original code https://github.com/os-autoinst/os-autoinst-distri-example/blob/10683a4777aae56837fd0aaa789666914c5b71df/.github/workflows/isotovideo-action.yml#L19
only fails the pipeline because of the empty match from grep (i.e. after removing all ok results, there is no other record).

Comment (okurz) [reply .github/workflows/isotovideo-action.yml]: hm, you are right. That changed in a pull request from @Martchus some months ago. Then it should be

```
run: jq .result testresults/result-*.json | grep -v ok || (echo "Test modules failed" && exit 1)
```

right?

Comment (baierjan) [reply .github/workflows/isotovideo-action.yml]: I believe not. The success of `grep -v ok` means something else than ok was matched (and that should be a failure); on the other hand, if `grep -v ok` fails (with exit code 1, btw.) it means there is no other result than ok, so it is actually a success. 
So what we want to have is
```sh
if $(jq .result testresults/result-*.json | grep -v ok); then echo "Test modules failed"; exit 1; else exit 0; fi
```
My original suggestion `jq .result testresults/result-*.json | grep -v ok && { echo "Test modules failed" && exit 1; } || true` should be the same. The "only" issue here is, that it masks error in jq. If we want to solve that, we need to introduce a little more logic.

Maybe
```sh
test $(jq .result testresults/result-*.json | grep -v ok | wc -l) -eq 0 || { echo "Test modules failed"; exit 1; }
```
could be more readable?

**Review by baierjan — CHANGES_REQUESTED** [review]
I believe we want to rebase over #36 and see what is left here.

**Comment by b10n1k** [comment]
> I believe we want to rebase over #36 and see what is left here.

Do we even need it anymore?

**Comment by okurz** [comment]
@b10n1k I incorporated your great ideas to fix the CI jobs but at least https://github.com/os-autoinst/os-autoinst-distri-example/pull/33/commits/c3ab292cecd89fdb6eb4fd4ef0ef14c44e07c739 would be beneficial on top

**Comment by baierjan** [comment]
> @b10n1k I incorporated your great ideas to fix the CI jobs but at least [c3ab292](https://github.com/os-autoinst/os-autoinst-distri-example/commit/c3ab292cecd89fdb6eb4fd4ef0ef14c44e07c739) would be beneficial on top

Yes, also the action version change to mitigate the node.js deprecation warning would be nice.

**Review by baierjan — COMMENTED** [review]
Only a minor issue, the commit message now does not correspond with the changes inside the commit; otherwise LGTM

**Review by okurz — CHANGES_REQUESTED** [review]
Correct, "https://github.com/os-autoinst/os-autoinst-distri-example/pull/33/commits/84284447abbb7bde81cba06f9d28b59b4838a9d4" shouldn't say that it would fix isotovideo CI jobs

**Review by okurz — APPROVED** [review]

**Review by baierjan — APPROVED** [review]
